src/phy/mod.rs

```diff
@@ -132,6 +137,33 @@ pub use self::tracer::Tracer;
 ))]
 pub use self::tuntap_interface::TunTapInterface;
 
+/// An ID that can be used to uniquely identify a packet to a [`Device`],
+/// sent or received by that same [`Device`]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, PartialEq)]
+pub struct PacketId {
+    id: usize,
+}
+
+impl PacketId {
+    pub const fn none() -> Option<Self> {
+        None
+    }
+
+    pub fn id(&self) -> usize {
+        self.id
+    }
+
+    pub(crate) fn copy(&self) -> Self {
+        Self { id: self.id }
+    }
```

Comment:
I think we can make `PacketId` `Copy`. Having a unique value for the `PacketId` is up to the user IMO.

Reply:
I agree. Initially it made sense to me to have it be non-`Copy`, but given that the only way problems can arise is when the `id` rolls over, it doesn't make any practical sense.

I'll make `PacketId` copy and update all the other issues/changes that occurred because of it,